hcl/hcl.go

```diff
@@ -648,7 +656,9 @@ func (c Config) AbsDir() string { return c.absdir }
 
 // IsEmpty returns true if the config is empty, false otherwise.
 func (c Config) IsEmpty() bool {
-	return c.Stack == nil && c.Terramate == nil
+	return c.Stack == nil && c.Terramate == nil &&
+		c.Vendor == nil && len(c.Asserts) == 0 &&
+		len(c.Generate.Files) == 0 && len(c.Generate.HCLs) == 0
```

Comment:
Isn't the empty check missing Globals and Imported ? (if a config only has globals or just an import block etc).